integration_tests/integration/referral-application.spec.ts

```diff
@@ -78,6 +78,56 @@ context('SignIn', () => {
     // And my task should be marked as completed
     page.checkStatus('ap-type', 'Completed')
   })
+  it('Saves my answers', () => {
+    // Given I complete the questionaire
+    const page = ReferralApplicationTasklist.visit()
+    checkEligibility(page)
+    page.startSection('Select the type of AP Required')
+    const typeOfAp = Page.verifyOnPage(TypeOfAP)
+    typeOfAp.answerType('esap')
+    typeOfAp.saveAndContinue()
+    const esapReasons = Page.verifyOnPage(EsapReasons)
+    esapReasons.chooseReasons(['secreting', 'cctv'])
+    esapReasons.saveAndContinue()
+    const roomSearches = Page.verifyOnPage(RoomSearches)
+    roomSearches.answerItems(['weapons', 'drugs'])
+    roomSearches.answerAgencyRequest('no')
+    roomSearches.saveAndContinue()
+    const cctv = Page.verifyOnPage(CCTV)
+    cctv.answerCctvReasons(['assualt-staff'])
+    cctv.answerCctvAgencyRequest('no')
+    cctv.saveAndContinue()
+
+    Page.verifyOnPage(ReferralApplicationTasklist)
+
+    // When I go back to previously answered sections
+    // Then my previous answers are visible
+    page.startSection('Check if the person is eligible')
+    page.checkInputsChecked('referralReason', 'likely')
+    cy.go('back')
```

Comment:
Now that we're adding persistence outside of the "session", ought we to be a bit clearer in demonstrating how this work adds to the earlier session-based persistence? 

e.g. if we added an initial commit to this PR which showed the limits of the existing behaviour we're starting out with:

- When I fill in several steps
- And restart the questionnaire
- Then my answers are preserved
- When my session is destroyed
- Then my answers are no longer preserved

then the commits with the changes might follow on, with a final concluding commit like this one here which shows:

- When I fill in several steps
- And my session is destroyed
- And I restart the questionnaire
- Then my answers are preserved

I think it's the fact that we no longer rely on the session which is key change we want to highlight?

Would something of that sort work and be an improvement? Or have I got my wires crossed here?

Reply:
I think that would work! I will have a go at it... Definitely a good way of making clear what the changes are. I also like the idea of having the E2E tests involved from the start of the PR

Reply:
@pezholio I was hoping to log in/out but I currently can't do it locally